src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -3087,16 +3161,26 @@ class SqlSelect a r | a -> r, r -> a where
     -- | Number of columns that will be consumed.
     sqlSelectColCount :: Proxy a -> Int
 
-    -- | Transform a row of the result into the data type.
-    sqlSelectProcessRow :: [PersistValue] -> Either T.Text r
-
     -- | Create @INSERT INTO@ clause instead.
     sqlInsertInto :: IdentInfo -> a -> (TLB.Builder, [PersistValue])
     sqlInsertInto = throw (UnexpectedCaseErr UnsupportedSqlInsertIntoType)
 
+-- | (Internal) Class for mapping results coming from 'SqlQuery'
+-- into actual results.
+--
+-- This looks very similar to @RawSql@, and it is!  However,
+-- there are some crucial differences and ultimately they're
+-- different classes.
+class SqlSelectCols a => SqlSelect a r | a -> r where
```

Comment:
even better we gain the ability to say `instance SqlSelect (SqlExpr (Value a)) a` and have it not overlap with `instance SqlSelect (SqlExpr (Entity a)) (Entity a)`

Reply:
Oh, oh, oh! So you mean we won't need to unwrap `Value` anymore?!

On the one hand, hell yes, on the other hand, that's a breaking change 🤔

Reply:
```
type Value a = a
unValue :: Value a -> a
unValue = id
```

Reply:
`pattern Value a = a where Value a = a` may even make it good for matching, eg `foo (Value a, Value b) = (a, b)`

Reply:
unfortunately we cant uses a type synonym since we need `SqlExpr (Value a)` to be different from `SqlExpr (Entity a)` to avoid overlapping instances. Though i guess we could just say the value one is overlappable.

Reply:
Oh actually looks like the type role annotation for #305 lets us use a `type Value a = a`

Reply:
or not